### PR TITLE
complete 06-Get-commentsByID

### PR DESCRIPTION
**Pull request by moskgit**
This will complete 06 comments by article id

### CODE REVIEW

**Review by yalicee — CHANGES_REQUESTED** [review]
add a test: 
Status 200, valid ID, but has no comments responds with an empty array of comments

**Comment by moskgit** [comment]
I've now refactored to add the requested test. But, I have a confusion. The requested test is asking me to write 'Status 200, valid ID, but has no comments responds with an empty array of comments' 
I previously wrote '404 Not Found' for the same. 
I'm unable to do both at the same time. 
Therefore, I skipped the last '404' test. Please clarify or let me know if there's a way to get both passed at the same time.
Thank you.

**Review by johnhutcheon — CHANGES_REQUESTED** [review]
Hi Mohsin, thanks for adding the extra 200 test. There is a way to get both this and the 404 test to pass. Think of how you can do an extra check to see if the article exists in another function. If you get stuck, drop us an nchelp 🙂

File: models.js [hunk 1]
```diff
@@ -55,21 +55,29 @@ function selectCommentsByArticleId(articleId) {
...
+        .then(({ rows }) => {
+            if (rows.length === 0) {
+                //Use queries to check what's giving an empty array, before explicitly rejecting.
+                return checkWhatsGivingEmptyArray(articleId)
+                }
+                return rows;
+            });
+        }
+        
+        function checkWhatsGivingEmptyArray(article_id) {
```
Comment (roisinoreilly) [inline models.js:74]: this variable name could be a little bit clearer

**Review by roisinoreilly — APPROVED** [review]
nicely refactored, PR approved :)

**Pull request merged**